src/Symfony/Component/ExpressionLanguage/ExpressionLanguage.php

```diff
@@ -84,7 +84,11 @@ public function parse($expression, $names)
             return $expression;
         }
 
-        $key = $expression.'//'.implode('-', $names);
+        $keys = array_map(function($key, $value){
+            return '{'."$key:$value".'}';
+        }, array_keys($names), $names);
```

Comment:
well, currently, you are doing a weird dumping of keys as ``{a:b}-{c:B}``. This is where I suggest using json-encoding would be simpler.